src/librustdoc/clean/types.rs

```diff
@@ -645,19 +645,31 @@ impl Attributes {
     }
 
     crate fn has_doc_flag(&self, flag: Symbol) -> bool {
+        !self.get_doc_flags(flag).is_empty()
+    }
+
+    crate fn get_doc_flags(&self, flag: Symbol) -> Vec<ast::MetaItem> {
```

Comment:
It might be slightly more performant to use `Iterator`, but it would require either wrapping it in an `Option` or using `dyn Iterator`, and it seemed like pre-mature optimization.